.changes/unreleased/Under the Hood-20230122-215235.yaml

```
@@ -0,0 +1,6 @@
+kind: Under the Hood
+body: Small optimization on manifest parsing benefitting large DAGs
+time: 2023-01-22T21:52:35.549814+01:00
+custom:
+  Author: boxysean
+  Issue: "6073"
```

I created a stand-alone issue for this surgical change. That way the changelog doesn't accidentally imply that this PR completely solves #6073.

But I'm open to the idea of multiple PRs leading back to the same issue if that's a precedent we've already set (or one we want to start using).
```suggestion
  Issue: "6697"
```